### PR TITLE
[eclipse/xtext-core#1566] update to LSP4J 0.10.0

### PR DESCRIPTION
**Pull request by cdietrich**
[eclipse/xtext-core#1566] update to LSP4J 0.10.0
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

**Comment by cdietrich** [comment]
TODO: how to deal with deprecated "deprecated" on DocumentSymbol

File: org.eclipse.xtext.ide.tests/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -14,8 +14,8 @@ Require-Bundle: org.eclipse.xtext;visibility:=reexport,
  org.eclipse.core.runtime;bundle-version="3.13.0",
  org.eclipse.xtext.ide,
  org.junit;bundle-version="4.12.0",
- org.eclipse.lsp4j;bundle-version="[0.9.0,0.10.0)";resolution:=optional,
- org.eclipse.lsp4j.jsonrpc;bundle-version="[0.9.0,0.10.0)";resolution:=optional,
```
Comment (szarnekow) [inline org.eclipse.xtext.ide.tests/META-INF/MANIFEST.MF:18]: We should keep 0.9.0 as lower bound, don’t you think so?

Comment (cdietrich) [reply org.eclipse.xtext.ide.tests/META-INF/MANIFEST.MF:18]: am not sure about this. if so we cant implement any of the new api

Comment (szarnekow) [reply org.eclipse.xtext.ide.tests/META-INF/MANIFEST.MF:18]: As long as we don’t implement any of these, there shouldn’t be a problem

Comment (cdietrich) [reply org.eclipse.xtext.ide.tests/META-INF/MANIFEST.MF:18]: ok, but when i will live with the deprecations for symbol.depreacted for now
its hard to circumvent them

**Review by szarnekow — APPROVED** [review]

**Pull request merged**